cardano-api/src/Cardano/Api/Typed.hs

```diff
@@ -1565,6 +1565,100 @@ data MultiSigScript = RequireSignature (Hash PaymentKey)
                     | RequireMOf Int [MultiSigScript]
   deriving (Eq, Show)
 
+instance ToJSON MultiSigScript where
+  toJSON (RequireSignature payKeyHash) = String . Text.decodeUtf8 $ serialiseToRawBytesHex $ payKeyHash
+  toJSON (RequireAnyOf reqSigs) = object [ "any" .= map toJSON reqSigs ]
+  toJSON (RequireAllOf reqSigs) = object [ "all" .= map toJSON reqSigs ]
+  toJSON (RequireMOf reqNum reqSigs) = object [convertMofN .= map toJSON reqSigs]
+   where
+     convertMofN = Text.pack $ (show reqNum ++ "of" ++ show (length reqSigs))
+
+instance FromJSON MultiSigScript where
+  parseJSON = Aeson.withObject "MultiSigScript" $ \obj -> do
+    -- Parse "all" multisig objects
+    allReqHashes <- case HMS.lookup "all" obj of
+                       Just (Array vecAllReqHashes) ->
+                         return . Just . RequireAllOf $ gatherSignatures vecAllReqHashes
+                       _ -> return Nothing
+
+    -- Parse "any" multisig objects
+    anyReqHashes <- case HMS.lookup "any" obj of
+                       Just (Array vecAnyReqHashes) ->
+                         return . Just . RequireAnyOf $ gatherSignatures vecAnyReqHashes
+                       _ -> return Nothing
+
+    -- Parse "mOfn" multisig objects
+    mofN <- return $ filterMofN obj
+
+    case (allReqHashes, anyReqHashes, mofN) of
+      (Nothing, Nothing, Nothing) -> fail "Script does not contain \"all\", \"any\" \
+                                          \or \"mOfn\" JSON objects."
+      (Just mss, _, _) -> return mss
+      (_, Just mss, _) -> return mss
+      (_, _, Just mss) -> return mss
```

Comment:
Hmm, based on the compile error below, the above probably should have `Nothing` in place of `_`.

Reply:
Could write it this way perhaps.

```
case catMaybes [allReqHashes, anyReqHashes, mofN] of
  [mss] -> return mss
  [] -> fail "Script does not contain \"all\", \"any\" or \"mOfn\" JSON objects."
  _ -> fail "Please specify only one type of JSON multisig object."
```